src/java/htsjdk/samtools/DownsamplingIterator.java

```diff
@@ -74,12 +85,30 @@ public DownsamplingIterator setIncludeNoRefReads(final boolean includeNoRefReads
         return this;
     }
 
+    /**
+     * Sets the target accuracy of the downsampling iterator.  The value should be thought of as
+     * probability +/- accuracy.  So a value of 0.001 would instruct the downsampling iterator to
+     * attempt to guarantee at accuracy to within 0.1%.  The downsampler will need to buffer reads
+     * for 1/accuracy templates, so setting this to extremely small numbers is not advisable.
+     */
+    public DownsamplingIterator setTargetAccuracy(final double accuracy) {
+        if (accuracy >= 1 || accuracy <= 0) throw new IllegalArgumentException("Illegal value. Must be 0 < accuracy < 1");
```

Comment:
to avoid overflow, accuracy must be greater than 1/MAX_INT (this is also true for other reasons, but one could conceive of enough memory to hold 2x10^9 records in memory, and so perhaps it's worth noting, we've been having overflows in other parts of the code due to similar issues.)